### PR TITLE
[BugFix] Add stream synchronize  for nccl migration backend

### PR DESCRIPTION
**Pull request by KuilongCui**
1. bugfix for migration stream sync problem
2. add cupy install tool

after dst worker recv data from col.recv, it need to copy data to self.gpu_cache. Before copying, it must wait cuda stream in col.recv. As nccl migration backend use cupy cuda stream, in this pr change migration stream to cupy stream for synchronize.

### CODE REVIEW

File: llumnix/backends/vllm/worker.py [hunk 1]
```diff
@@ -145,3 +146,14 @@ def shutdown(self) -> None:
     def restart(self) -> None:
         self.init_model()
         self.init_cache_engine(self.cache_config)
+
+    # Note: this Method is only used for testing !!!
+    def set_gpu_cache(self, data):
```
Comment (ZeldaHuang) [inline llumnix/backends/vllm/worker.py:151]: We should implements  these functions in a `TestClass`

File: tests/backends/vllm/test_migration_backend.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+from vllm.engine.arg_utils import EngineArgs
+from vllm.utils import get_distributed_init_method, get_ip, get_open_port
+
+from llumnix.arg_utils import EngineManagerArgs
+from llumnix.utils import random_uuid
+
+from tests.backends.vllm.test_worker import create_worker
+
+@pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Need at least 2 GPU to run the test.")
```
Comment (ZeldaHuang) [inline tests/backends/vllm/test_migration_backend.py:27]: `torch.cuda.device_count() < 2`?

File: tests/backends/vllm/test_worker.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    migration_cache_size = ray.get(worker.execute_method.remote('reserve_memory_for_migration',
+                                                                migration_config=migraiton_config,
+                                                                model_config=engine_config.model_config,
+                                                                cache_config=engine_config.cache_config,
+                                                                parallel_config=engine_config.parallel_config))
+    assert migration_cache_size == occupy_memory
+
+    ray.shutdown()
+
+@pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Need at least 2 GPU to run the test.")
```
Comment (ZeldaHuang) [inline tests/backends/vllm/test_worker.py:86]: `torch.cuda.device_count() < 2`?

Comment (KuilongCui) [reply tests/backends/vllm/test_worker.py]: get

File: tests/backends/vllm/test_migration_backend.py (hunk 2 above)
Comment (KuilongCui) [reply tests/backends/vllm/test_migration_backend.py]: get

File: llumnix/backends/vllm/worker.py (hunk 1 above)
Comment (KuilongCui) [reply llumnix/backends/vllm/worker.py]: get
see https://github.com/AlibabaPAI/llumnix/pull/22

File: tests/backends/vllm/test_migration_backend.py (hunk 2 above)
Comment (KuilongCui) [reply tests/backends/vllm/test_migration_backend.py]: see https://github.com/AlibabaPAI/llumnix/pull/22

File: tests/backends/vllm/test_worker.py (hunk 3 above)
Comment (KuilongCui) [reply tests/backends/vllm/test_worker.py]: see https://github.com/AlibabaPAI/llumnix/pull/22

**Review by zhypku — APPROVED** [review]

**Review by ZeldaHuang — APPROVED** [review]

**Pull request merged**